traffic_ops/traffic_ops_golang/cdni/telemetry.go

```diff
@@ -0,0 +1,112 @@
+package cdni
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/lib/pq"
+
+	"github.com/apache/trafficcontrol/traffic_ops/traffic_ops_golang/api"
+)
+
+func GetTelemetries(inf *api.APIInfo, ucdn string) (Capabilities, error) {
+	capRows, err := inf.Tx.Tx.Query(CapabilityQuery, FciTelemetry, ucdn)
+	if err != nil {
+		return Capabilities{}, fmt.Errorf("querying capabilities: %w", err)
+	}
+	defer capRows.Close()
+	capabilities := []CapabilityQueryResponse{}
+	for capRows.Next() {
+		var capability CapabilityQueryResponse
+		if err := capRows.Scan(&capability.Id, &capability.Type, &capability.UCdn); err != nil {
+			return Capabilities{}, fmt.Errorf("scanning db rows: %w", err)
+		}
+		capabilities = append(capabilities, capability)
+	}
+
+	fciCaps := Capabilities{}
+
+	for _, cap := range capabilities {
+		fciCap := Capability{}
+		footRows, err := inf.Tx.Tx.Query(FootprintQuery, cap.Id)
```

Comment:
Similar to a prior comment, is it possible to come up with a single query here (or a constant number of queries) that returns all the data we need instead of making a lot of small queries?